### PR TITLE
[SPARK-14806] [SQL] Alias original Hive options in Spark SQL conf

### PR DESCRIPTION
**Pull request by bomeng**
## What changes were proposed in this pull request?

There are couple options we should alias: `spark.sql.variable.substitute` and `spark.sql.variable.substitute.depth`.

The Hive config options are `hive.variable.substitute` and `hive.variable.substitute.depth`

Since we have `spark.sql.variable.substitute` as an alias of `hive.variable.substitute`, we will use it for the `SET` command.
## How was this patch tested?

Test was added to the existing one to test this new feature.

### CODE REVIEW

**Comment by rxin** [comment]
You are not aliasing hive.variable.substitute to spark.sql.variable.substitute here, are you?

**Comment by bomeng** [comment]
@rxin Just wanna confirm, you want to let user to do `SET hive.variable.substitute=true/false` in SQL? It will logWarning in `setConfWithCheck()` method and I just convert it there?

**Comment by rxin** [comment]
yup.

File: sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -471,12 +471,24 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val HIVE_VARIABLE_SUBSTITUTE_ENABLED =
+    SQLConfigBuilder("hive.variable.substitute")
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: i don't think we shoul define them here

**Comment by rxin** [comment]
@bomeng maybe i wasn't clear. the goal is:

whenever hive.variable.substitute is set by the user, just set the value of hive.variable.substitute; whenever hive.variable.substitute is read, just read the value of hive.variable.substitute.

**Comment by bomeng** [comment]
I think you mean set the value of `spark.sql.variable.substitute` and read `spark.sql.variable.substitute` above. I will post another try shortly.

**Comment by rxin** [comment]
Hi @bomeng  - thanks for doing this. I thought about it more and actually think maybe it's not worth it to support these aliases, because it is going to be confusing when users do get all configs. Also it makes it sound like we can support all the Hive options, when in reality we don't. So unfortunately I think we should just close this one for now. What do you think?

**Comment by bomeng** [comment]
@rxin I am open to your decision. I think it is still useful to allow user to use "SET" command by using spark.sql.variable.substitute as configuration. Currently, the "SET" command does not support that. Part of my codes is for fixing it, do you think it is still valid for that part?

**Comment by bomeng** [comment]
closing it. thanks.

**Pull request closed without merging**